packages/web-components/fast-components/src/tabs/tabs.ts

```diff
@@ -0,0 +1,324 @@
+        if (this.activeId !== undefined) {
+            return this.tabIDs.indexOf(this.activeId) === -1
+                ? 0
+                : this.tabIDs.indexOf(this.activeId);
+        } else {
+            return 0;
+        }
+    }
+
+    private getTabs = (): void => {
+        this.tabIDs = this.getTabIds();
+        this.tabPanelIDs = this.getTabPanelIds();
+        this.activeTabIndex = this.getActiveIndex();
+        this.tabs.forEach((tab: HTMLElement, index: number) => {
+            if (tab.slot === "tab") {
+                tab.setAttribute(
+                    "id",
+                    this.tabIDs[index] === undefined || this.tabIDs[index] === null
+                        ? `tab-${index + 1}`
+                        : this.tabIDs[index]
+                );
+                tab.setAttribute(
+                    "aria-selected",
+                    this.activeTabIndex === index ? "true" : "false"
+                );
+                tab.setAttribute(
+                    "aria-controls",
+                    this.tabPanelIDs[index] === undefined ||
+                        this.tabPanelIDs[index] === null
+                        ? `panel-${index + 1}`
+                        : this.tabPanelIDs[index]
+                );
+                tab.setAttribute(
+                    "style",
+                    this.isHorizontal()
+                        ? `grid-column: ${index + 1};`
+                        : `grid-row: ${index + 1};`
+                );
+                tab.addEventListener("click", this.handleTabClick);
+                tab.addEventListener("keydown", this.handleTabKeyDown);
+                tab.setAttribute("tabindex", this.activeTabIndex === index ? "0" : "-1");
+                if (this.activeTabIndex === index) {
+                    this.activeTab = tab;
+                }
+                !this.isHorizontal()
+                    ? tab.classList.add("vertical")
+                    : tab.classList.remove("vertical");
+            }
+        });
+    };
+
+    private getTabPanels = (): void => {
+        this.tabIDs = this.getTabIds();
+        this.tabPanelIDs = this.getTabPanelIds();
+        this.tabPanels.forEach((tabpanel: HTMLElement, index: number) => {
+            tabpanel.setAttribute(
+                "id",
+                this.tabPanelIDs[index] === undefined || this.tabPanelIDs[index] === null
+                    ? `panel-${index + 1}`
+                    : this.tabPanelIDs[index]
+            );
+            tabpanel.setAttribute(
+                "aria-labeledby",
+                this.tabIDs[index] === undefined || this.tabIDs[index] === null
+                    ? `tab-${index + 1}`
+                    : this.tabIDs[index]
+            );
+            this.activeTabIndex !== index
+                ? tabpanel.setAttribute("hidden", "")
+                : tabpanel.removeAttribute("hidden");
+        });
+    };
+
+    private getTabIds(): string[] {
```

Comment:
Is there a case where these ID's wont exist? If so, I'd prefer to see an accurate type signature for these functions and not coercing with `as string`.